src/contributing/prs.md

```diff
@@ -0,0 +1,150 @@
+and reviewer; and, once agreement is reached, the pull request will be merged.
+
+This page explains yk's approach to the pull request model, based on
+observations of good practise in other projects, and our own experiences.
+Please note that this is a living process: the details are subject to continual
+refinement and change, and no document can perfectly capture all the
+possibilities that will crop up in real life.
+
+
+## What makes a good pull request?
+
+The typical aim of a pull request to a yk repository is to make a focussed
+change such as: add a new feature; fix a bug; or edit documentation. Good pull
+requests are:
+
+  1. *Self-contained*. A pull request should not try and change everything at
+     once. Ideally a pull request fixes a single bug or adds a single feature;
+     but sometimes a set of interdependent fixes or features makes more sense.
+     In general, small pull requests are better than big pull requests.
+
+  2. *Tested*. Every piece of new functionality must have at least one
+     accompanying test. Authors should also strive not to break any other part
+     of the system. yk runs various tests before a pull request is merged: this
+     minimises, but does not remove, the chances of a merged pull request
+     breaking the repository: careful thought and testing on the part of the
+     pull request author are still required.
+
+  3. *Documented*. Code diffs show *what* has been changed, but not *why*.
+     Documentation explains to other humans the context for a change (i.e. why
+     something needed to be changed), the reason why the change is as it is
+     (i.e. could the change have taken a different form? what alternatives were
+     considered or attempted?), and the consequences of doing so (i.e. does
+     this make something easier or harder in the future?). Documentation comes
+     in the form of comments in code, “external” documentation (such as you
+     read here), and commit and pull request messages. In whatever form it
+     comes, documentation must be clear (as easy to understand as possible),
+     concise (as short as possible while getting all the required points
+     across), and complete (not missing any important points).
+
+  4. *Useful*. The more people that will benefit from a pull request, the more
+     likely it is to be merged. The bar for “useful” is set fairly low, but
+     that does not mean that every pull request should be merged: for example,
+     a pull request which simply changes code to your preferred style is not
+     only not useful but may cause problems for other people working in
+     parallel. However, porting the code to a little used platform probably is
+     often useful (provided it doesn’t cause undue problems for more widely
+     used platforms).
+
+
+## The pull request process in yk repositories
+
+To raise a pull request, you must:
+
+  1. Fork the relevant yk GitHub repository to your account and clone it to
```

Comment:
I'd remove:

> and clone it to your local computer.

If it's a trivial typo fix, the contributor might do it all online in the GH web UI.

Reply:
Fixed in 6682683.